presentation/src/main/java/org/cryptomator/presentation/presenter/BrowseFilesPresenter.kt

```
@@ -1032,21 +1030,35 @@ class BrowseFilesPresenter @Inject constructor( //
-	@Callback
+	@Callback(dispatchResultOkOnly = false)
 	fun selectedFiles(result: ActivityResult) {
+		if (!result.isResultOk) {
+			releasePickerLease()
+			return
+		}
 		val fileUris = getFileUrisFromIntent(result.intent())
+		fileUris.forEach { uri ->
+			try {
+				context().contentResolver.takePersistableUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION)
+			} catch (e: SecurityException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)